jupyter_server/auth/identity.py

```diff
@@ -81,6 +81,39 @@ def fill_defaults(self):
         if not self.display_name:
             self.display_name = self.name
 
+    def to_string(self) -> str:
+        """Serialize a user to a string for storage in a cookie
+        If overriding in a subclass, make sure to define from_string as well.
+        Default is just the user's username.
+        """
+        # default: username is enough
+        cookie = json.dumps(
+            {
+                "username": self.username,
+                "name": self.name,
+                "display_name": self.display_name,
+                "initials": self.initials,
+                "color": self.color,
+            }
+        )
+        return cookie
+
+    @classmethod
+    def from_string(cls, serialized_user: str) -> User:
+        """Inverse of user_to_cookie"""
+        user = json.loads(serialized_user)
+        try:
```

Comment:
Is the try/except necessary?  Since `loads` can raise as well, we might as well let things fly (no try/except) unless we want to introduce a custom exception.